### PR TITLE
compatibility with simgrid 3.20

### PR DESCRIPTION
**Pull request by frs69wq**

### CODE REVIEW

**Comment by rafaelfsilva** [comment]
@frs69wq, Does SimGrid 3.20 keeps backwards compatibility with the function names for 3.19?

@henricasanova, In case it does not keep backwards compatibility (see question above for Fred), shall we make only v1.1 compatible with SimGrid 3.20, and then limit v1.0.1 to 3.19, or should we say that both new releases will be compatible with SimGrid 3.20? I tend to move with the later, but comments are welcome.

**Comment by henricasanova** [comment]
I believe Martin told me that there were name changes that would not be backward compatible. I'll let Fred confirm. I think that making both releases compatible with SimGrid 3.20 is very easy. Most calls are in on part of the code. To mean, moving from 1.0 to 1.0.1 can absolutely entail upgrading to the SimGrid 3.20 API.  I'm installing SimGrid 3.20 right now and I'll commit to the 1.0 branch something that makes it all work, merging with master.

**Comment by frs69wq** [comment]
@mquinson may correct me, but all the changes to the API with the snake_case madness were made in a way that codes using the  camelCase function names continue to work ... for now. Compilation should display a lot of warnings telling the user to modify his/her code to use the snake_case version because camelCase wrappers will be dropped in 3.23 (1 year from now). 

None of the functions used by wrench is part of the non-backward compatible set mentioned by @mquinson. So, it should be fine. 

The one thing that may break though is that the MIN/MAX macros are not exposed anymore, but in a C++ code as wrench, relying on std::min and std::max is the way to go. 

To summarize, you can merge this PR whenever you want, there is no hurry, wrench will still work without this PR and with 3.20 (except for this MIN/MAX point). I made the changes on the fly each time a new class were snake_cased.

**Comment by henricasanova** [comment]
Just an update. I am having compilation problems with 3.20 on Mac (the DEPRECATED macros don't compile with clang with error messages like "expected argument" etc.)  I have no time to investigate right now. So Rafael I suggest we stick to 3.19 until we all see each other in Rennes.  I have, however, pushed a commit to replace all MAX/MIN by std::max/std::min, as suggested by Fred.

**Comment by mquinson** [comment]
Hello guys,
the exact error message of clang about DEPRECATED macro would be really welcome. We did not change them since 3.18, I think, and we do test on mac. That's surprising.

As for the changes, I glanced through the patch provided by Fred, and I confirm that none of them are mandatory for 3.20 if your users can live with tons and tons (and tons) of deprecation warnings. 

Just out of curiosity, what would incite you to support both SimGrid v3.19 and v3.20?

And as a final note, sorry for these API changes. We definitely plan to keep them rare, but that cleanup was really mandatory.

**Comment by henricasanova** [comment]
The error messages I was getting where because of an old xbt folder in /opt/local/include on my machine that came from some previous bogus SimGrid install. Sorry for the noise.   

This morning, without looking at this thread, I created a branch 1.0_SimGrid3.20  in which I did the port from 3.19 to 3.20.  But I just realized that Fred had done a pull request where he also did this (somehow I missed that e-mail?).   I also see that his request has conflicts... So I am not sure how to proceed. What we're doing right now is developing in 1.0 and merging 1.0 into master for thing that should apply for release 1.0.1.     I didn't do that because I don't think our Travis setup does SimGrid 3.20 right now, and I didn't want all tests to fail.

The simplest right now is to merge my 1.0_SimGrid3.20 branch into the 1.0 branch, and then merge the 1.0 branch into the master branch...  Another option is to take Fred's pull request, resolve the conflicts, and then apply it to branch 1.0, and then merge it into master.

@rafael any thoughts?

**Comment by frs69wq** [comment]
well, there was no email about this PR, but this discussion you’re into is attached to this very same PR, so I assumed you saw it ;-)
There was no conflict at the time at submit it, but as the code of wrench is moving fast, some may have appeared. 
If you made the port in your branch, you can just discard this PR, I won’t hold any grudge for that.

**Comment by henricasanova** [comment]
Yeah, I am doing way too many things at once, which has become a problem in my spacing out all over the place. And yes, it had occurred to me the conflicts had appeared. We'll likely discard your PR then.... sorry about the miscommunication.

**Comment by henricasanova** [comment]
Ended up merging my 1.0_SimGrid3.20 branch into 1.0, and then 1.0 into master. Sorry again for causing Fred to do what ended up being redundant work because I missed this PR somehow.
